### ISSUE TITLE
github login failed every time

### ISSUE THREAD

**Issue by tsgsz** [microsoft/vscode#160649]
Type: <b>Bug</b>

The login step in browser is good, and successfull, but when redirect to vscode, vscode can also recive the redirect request, because it will pop up a window to ask me if allow it, but after that, it stuck until timeout

VS Code version: Code 1.71.0 (784b0177c56c607789f9638da7b6bf3230d47a8c, 2022-09-01T07:36:10.600Z)
OS version: Windows_NT x64 10.0.22000
Modes:
Sandboxed: No
Remote OS version: Linux x64 3.10.0-1127.19.1.el7.x86_64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (4 x 3300)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.91GB (24.16GB free)|
|Process Argv|--crash-reporter-id a72f7270-60dd-4eec-8d30-1feb1135c48c|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: 阿里云|
|OS|Linux x64 3.10.0-1127.19.1.el7.x86_64|
|CPUs|Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz (2 x 2499)|
|Memory (System)|7.64GB (4.40GB free)|
|VM|25%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-docker|ms-|1.22.1
vscode-language-pack-zh-hans|MS-|1.71.9070915
python|ms-|2022.14.0
vscode-pylance|ms-|2022.9.10
jupyter|ms-|2022.8.1002431955
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.9
remote-ssh|ms-|0.84.0
remote-ssh-edit|ms-|0.80.0
remote-wsl|ms-|0.66.3
vim|vsc|1.23.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411cf:30557515
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
pylanb8912:30545647
vsclangdf:30486550
c4g48928:30535728
hb751961:30553087
dsvsc012:30540252
azure-dev_surveyone:30548225
i497e931:30553904

```

</details>

**Comment by tsgsz** [comment]
And I have tried win11, win10, MACOS, also with proxy or not with proxy, nothing helped

**Comment by tsgsz** [comment]
Any update ?

**Comment by TylerLeonhardt** [comment]
Sorry could you attach a video or a gif? That would really help me understand what you're seeing.

/gifPlease

**Comment by VSCodeTriageBot** [comment]
Thanks for reporting this issue! Unfortunately, it's hard for us to understand what issue you're seeing. Please help us out by providing a screen recording showing exactly what isn't working as expected. While we can work with most standard formats, `.gif` files are preferred as they are displayed inline on GitHub. You may find https://gifcap.dev helpful as a browser-based gif recording tool.

If the issue depends on keyboard input, you can help us by enabling screencast mode for the recording (`Developer: Toggle Screencast Mode` in the command palette).

Happy coding!

**Comment by ronakay22** [comment]
How do I make y'all stop sending me emails. I don't even do git hub

**Comment by tsgsz** [comment]
> /gif
![GIF 2022-9-21 星期三 23-58-30](https://user-images.githubusercontent.com/990478/191554230-0c482f51-d911-40b9-a909-3ecbabeb1c76.gif)
sorry for the gif delay

**Comment by TylerLeonhardt** [comment]
can you provide the contents of the GitHub Authentication output pane:
![image](https://user-images.githubusercontent.com/2644648/128087382-c706864f-7005-4a2e-878b-662c69c4457e.png)

And also anything that shows up in Dev Tools:

* ctrl + shift + P
* "Dev Tools"

**Comment by tsgsz** [comment]
> can you provide the contents of the GitHub Authentication output pane: ![image](https://user-images.githubusercontent.com/2644648/128087382-c706864f-7005-4a2e-878b-662c69c4457e.png)
> 
> And also anything that shows up in Dev Tools:
> 
> * ctrl + shift + P
> * "Dev Tools"

![0751fbe21786314d50f2a02646c4862](https://user-images.githubusercontent.com/990478/192132735-bd6869b7-b07e-4248-952f-4f30d6d91147.png)
Nothing else

**Comment by TylerLeonhardt** [comment]
Very interesting... you see the spinning `sign in to github.com...` in your status bar... can you click on that and click cancel and then it'll ask you via a notification if you want to try logging in a different way... can you try that? Let me know if you need a video to understand what I'm talking about.

**Comment by tsgsz** [comment]
> Very interesting... you see the spinning `sign in to github.com...` in your status bar... can you click on that and click cancel and then it'll ask you via a notification if you want to try logging in a different way... can you try that? Let me know if you need a video to understand wh

at I'm talking about.

Already tried so many times， here is the result ![GIF 2022-10-2 星期日 10-58-25](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif)

**Comment by tsgsz** [comment]
> > Very interesting... you see the spinning `sign in to github.com...` in your status bar... can you click on that and click cancel and then it'll ask you via a notification if you want to try logging in a different way... can you try that? Let me know if you need a video to understand what I'm talking about.
> 
> Already tried so many times， here is the result ![GIF 2022-10-2 星期日 10-58-25](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) [ ![GIF 2022-10-2 星期日 10-58-25](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) ](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) [ ](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif)

Hi, is there any update?

**Comment by TylerLeonhardt** [comment]
@tsgsz let's try to get you unblocked first... can you open up Windows Credential Manager and go to Windows Credentials
![image](https://user-images.githubusercontent.com/2644648/195432533-a66df591-a2ee-418c-9944-da8f87c98de3.png)

Then look for any credentials that start with `vscode` and delete them. Then try logging in again.

**Comment by tsgsz** [comment]
> @tsgsz let's try to get you unblocked first... can you open up Windows Credential Manager and go to Windows Credentials ![image](https://user-images.githubusercontent.com/2644648/195432533-a66df591-a2ee-418c-9944-da8f87c98de3.png)
> 
> Then look for any credentials that start with `vscode` and delete them. Then try logging in again

There is no  credentials that start with vscode, I delete all of the rest credentials and loggin again, the result is all the same.

I really don't think it's the credential problem, because couple of days ago, I try it in my mac, didn't work either, and my windows also resinstalled once, nerver work.

**Comment by TylerLeonhardt** [comment]
Let's see if this is a keytar issue then... first log in in vscode (even though it fails)

if you have nodejs installed, you can run this simple one-liner to see if the problem is how your keychain is setup:

```
npx @emacs-grammarly/keytar-cli find-creds -s vscodevscode.github-authentication
```
or if you're on Insiders:
```
npx @emacs-grammarly/keytar-cli find-creds -s vscode-insidersvscode.github-authentication
```
This is using the following open source package: https://github.com/emacs-grammarly/keytar-cli

What you should get is something like this:
```js
[
  {
    account: 'github.auth',
    password: '....... longggg string ......'
  }
]
```
> NOTE: on macOS I got a system prompt to allow this process to access the keychain... that may or may not be the case for you

Additionally, we can see if we can properly set passwords:

```
npx @emacs-grammarly/keytar-cli set-pass testingvscode testingvscode testingvscode
```
and then
```
npx @emacs-grammarly/keytar-cli find-creds -s testingvscode
```
should give us that newly created password.

**Comment by tsgsz** [comment]
> Let's see if this is a keytar issue then... first log in in vscode (even though it fails)
> 
> if you have nodejs installed, you can run this simple one-liner to see if the problem is how your keychain is setup:
> 
> ```
> npx @emacs-grammarly/keytar-cli find-creds -s vscodevscode.github-authentication
> ```
> 
> or if you're on Insiders:
> 
> ```
> npx @emacs-grammarly/keytar-cli find-creds -s vscode-insidersvscode.github-authentication
> ```
> 
> This is using the following open source package: https://github.com/emacs-grammarly/keytar-cli
> 
> What you should get is something like this:
> 
> ```js
> [
>   {
>     account: 'github.auth',
>     password: '....... longggg string ......'
>   }
> ]
> ```
> 
> > NOTE: on macOS I got a system prompt to allow this process to access the keychain... that may or may not be the case for you
> 
> Additionally, we can see if we can properly set passwords:
> 
> ```
> npx @emacs-grammarly/keytar-cli set-pass testingvscode testingvscode testingvscode
> ```
> 
> and then
> 
> ```
> npx @emacs-grammarly/keytar-cli find-creds -s testingvscode
> ```
> 
> should give us that newly created password.

`npx @emacs-grammarly/keytar-cli find-creds -s vscodevscode.github-authentication` this return "[]"
`npx @emacs-grammarly/keytar-cli set-pass testingvscode testingvscode testingvscode` this is normal

and I manually set vscodevscode.github-authentication using the set-pass command, the output keytar is ok, just vsocde still can't login

**Comment by tsgsz** [comment]
> 

My friend, any clue?

**Comment by tsgsz** [comment]
Hi， is there anyone still working on this issue？

**Comment by TylerLeonhardt** [comment]
I just got back from vacation and am juggling multiple items. This issue hasn't been forgotten

**Comment by tsgsz** [comment]
> I just got back from vacation and am juggling multiple items. This issue hasn't been forgotten

thanks

**Comment by TylerLeonhardt** [comment]
@tsgsz can you run these two lines in PowerShell and tell me what you get?

```
Invoke-RestMethod -Method Post https://vscode.dev/codeExchangeProxyEndpoints/github/login/oauth/access_token
Invoke-RestMethod -Method Post https://vscode.dev/codeExchangeProxyEndpoints/github/login/oauth/access_token -Body 'code=123'
```

They will fail, but I want to make sure you at least get a response from the server

**Comment by tsgsz** [comment]
result as follow:
  
![03799d05ee50ac0bbd50e52c5969ee5](https://user-images.githubusercontent.com/990478/200117837-1eeb01c2-8025-4332-b61a-9344a16faeb1.png)

the text version:
Invoke-RestMethod : No code property provided in body
+ Invoke-RestMethod -Method Post https://vscode.dev/codeExchangeProxyEn ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod]，WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand

Invoke-RestMethod : bad_verification_code: The code passed is incorrect or expired. (https://docs.github.com/apps/managing-oauth-apps/troubleshooting-oauth-app-access-token-request-errors/#bad-verification-code)
+ Invoke-RestMethod -Method Post https://vscode.dev/codeExchangeProxyEn ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod]，WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand

**Comment by tsgsz** [comment]
knock knock

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by melMass** [comment]
Hi,

I've just read the full issue and I have the exact same issue.
I'm using ssh keys to login to github, this works everywhere except in vscode. It does popup the browser login window
![image](https://user-images.githubusercontent.com/7041726/208484548-628534d3-024e-48ea-8edd-12d248dcf0e8.png)

But clicking signing with browser just goes to a blank `localhost` page most of the time, once it did show me a successfully login you can go back to the app, but in vscode it was still complaining.

`npx @emacs-grammarly/keytar-cli find-creds -s vscode-insidersvscode.github-authentication`
Returns 
```json
[
  {
    account: 'github.auth',
    password: 'AQxlmzCMdedeFgtEgtjogfetgwt/lfsAefgAgteagt3geugtYerSgetOYrDD\n' +
      'Blablablabal(removed)'
  }
]
```

And this is `Github Auth` Log:
![image](https://user-images.githubusercontent.com/7041726/208484954-714696dc-adf2-4d48-af84-2188da365e82.png)

This only happens on my windows machine and funny to note that while trying to solve this, every single commits had a different `user`... not sure why but it might be related:
![image](https://user-images.githubusercontent.com/7041726/208485386-d1c42ae3-18d1-4503-87ae-805f12880d01.png)

**Comment by TylerLeonhardt** [comment]
That doesn't seem to be related. VS Code does not open those popups. I got a new issue going for you.

**Comment by tsgsz** [comment]
I've got a new condition, my last computer naturally ok for days ago(I've done nothing, and no update), but my new mac can't login, exactly the same behavior, which remind me maybe there is some token or other thing time related?

**Comment by TylerLeonhardt** [comment]
Hey @tsgsz, hope you're feeling better. COVID sucks. For macOS, how did you install it. Could you record a video of you going through the flows? I'd love it if you could try the fallback flows to see if any work for you.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by tsgsz** [comment]
I have 2 macs, one is new, so I can't use setting sync and code pilot, but one is old, it use to login successfully in setting sync, but now can't login with pilot,  which looks like this
<img width="371" alt="截屏2023-01-21 上午9 59 00" src="https://user-images.githubusercontent.com/990478/213836473-a12c892a-e74e-484e-9d32-0ef065d577b7.png">
This is interesting now, I have 3 computers, 2 mac, 1 pc,  pc now can do everything(I did nothing actually, it just became fine one day), one mac can sync settings but no code pilot, one mac can do nothing...

**Comment by tsgsz** [comment]
> 

the video you want is exactly like this except it's in mac

**Comment by tsgsz** [comment]
> > Very interesting... you see the spinning `sign in to github.com...` in your status bar... can you click on that and click cancel and then it'll ask you via a notification if you want to try logging in a different way... can you try that? Let me know if you need a video to understand what I'm talking about.
> 
> Already tried so many times， here is the result ![GIF 2022-10-2 星期日 10-58-25](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) [ ![GIF 2022-10-2 星期日 10-58-25](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) ](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif) [ ](https://user-images.githubusercontent.com/990478/193435966-d06d976f-b32a-490f-9d33-73d1cbe93fab.gif)

The video you want is exaclty the same with this, but in mac

**Comment by tsgsz** [comment]
I check the "can do nothing" mac's key chain, compare to the "can do one thing" mac, the result like below
<img width="886" alt="截屏2023-01-21 上午10 08 43" src="https://user-images.githubusercontent.com/990478/213838877-13ad9f7e-2bbe-48b4-8d20-62d83232b380.png">

**Comment by dineshyadav3169** [comment]
same issue

**Comment by sketing** [comment]
same issue

**Comment by lrlunin** [comment]
same issue

**Comment by TylerLeonhardt** [comment]
The "same issue" comments... please just use the `+1` reaction feature of GitHub. We call this out in our wiki:
https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions#before-submitting-an-issue

A new version of VS Code will be going out today that has a fix that might help in this space. Can you get it and tell me how it goes?

**Comment by tsgsz** [comment]
> The "same issue" comments... please just use the `+1` reaction feature of GitHub. We call this out in our wiki: https://github.com/microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions#before-submitting-an-issue
> 
> A new version of VS Code will be going out today that has a fix that might help in this space. Can you get it and tell me how it goes?

just tried, nothing changed

**Comment by ThrowMeForALoop** [comment]
same issue on windows

**Comment by tsgsz** [comment]
Any update? or any clue?

**Comment by tsgsz** [comment]
I have one pc, and one mac to be good now,  only one mac(but my mac in company doesn't work), hope before it became ok, I can help u find the bug...

**Comment by TylerLeonhardt** [comment]
I had another idea. My goal is the try to understand where you get stuck.

Run vscode in a terminal with the flag `--disable-keytar`.

If you are able to complete the login flow, then the issue is talking to the keychain on your OS.

This disables persistent auth sessions so reloading the window will throw out your login session but at least we'll have more information.

**Comment by tsgsz** [comment]
> I had another idea. My goal is the try to understand where you get stuck.
> 
> Run vscode in a terminal with the flag `--disable-keytar`.
> 
> If you are able to complete the login flow, then the issue is talking to the keychain on your OS.
> 
> This disables persistent auth sessions so reloading the window will throw out your login session but at least we'll have more information.

got this: 
![image](https://user-images.githubusercontent.com/990478/224594919-40ccbdf3-62a2-4052-a185-c04282026ebb.png)

**Comment by TylerLeonhardt** [comment]
Now _that's_ unusual. I don't repro that. Also, my error is different when I run with a command that isn't real:

![Image](https://user-images.githubusercontent.com/2644648/224615861-34c915df-09f8-4ac9-8c28-992ed938d26f.png)

This is the one on your Mac, yes? My `code` comes from `/usr/local/bin/code` (running `which code`)... what does yours say? Also what does `code --help` and `code --version` output.

**Comment by tsgsz** [comment]
> Now _that's_ unusual. I don't repro that. Also, my error is different when I run with a command that isn't real:
> 
> ![Image](https://user-images.githubusercontent.com/2644648/224615861-34c915df-09f8-4ac9-8c28-992ed938d26f.png)
> 
> This is the one on your Mac, yes? My `code` comes from `/usr/local/bin/code` (running `which code`)... what does yours say? Also what does `code --help` and `code --version` output.

my vscode is the latest version in office website, mac osx intel, is  that the same?

**Comment by tsgsz** [comment]
> Now _that's_ unusual. I don't repro that. Also, my error is different when I run with a command that isn't real:
> 
> ![Image](https://user-images.githubusercontent.com/2644648/224615861-34c915df-09f8-4ac9-8c28-992ed938d26f.png)
> 
> This is the one on your Mac, yes? My `code` comes from `/usr/local/bin/code` (running `which code`)... what does yours say? Also what does `code --help` and `code --version` output.

![image](https://user-images.githubusercontent.com/990478/224635086-9afa271b-734a-4255-82cd-6bd1f747040f.png)

yes, this is the one on my mac

**Comment by TylerLeonhardt** [comment]
Apparently, this flag is not available in the standalone CLI but is in the normal CLI... anyway, another way to do this is to run:
```
Preferences: Configure Runtime Arguments
```
and add a `"disable-keytar": true`

it will say it's not allowed, but it is. Ignore the warning and restart (kill and start) VS Code.

**Comment by tsgsz** [comment]
> Apparently, this flag is not available in the standalone CLI but is in the normal CLI... anyway, another way to do this is to run:
> 
> ```
> Preferences: Configure Runtime Arguments
> ```
> 
> and add a `"disable-keytar": true`
> 
> it will say it's not allowed, but it is. Ignore the warning and restart (kill and start) VS Code.

find a way to do it anyway, disable succeed, like this:
![image](https://user-images.githubusercontent.com/990478/224877901-e014d142-de9b-4618-a043-5c186f7f8b09.png)

But the problem still the same, nothing changed, like this:
![image](https://user-images.githubusercontent.com/990478/224878109-b32ed25d-6f34-4af3-bf4c-79ed74cc22d8.png)

**Comment by tsgsz** [comment]
After disable the keytar,  everything is exactly the same like before

**Comment by TylerLeonhardt** [comment]
@tsgsz just to be perfectly sure... with keytar disabled, can you share the contents of the `Main` output channel?

**Comment by tsgsz** [comment]
> @tsgsz just to be perfectly sure... with keytar disabled, can you share the contents of the `Main` output channel?

like this
![image](https://user-images.githubusercontent.com/990478/226315681-9220a460-6f4b-4fd8-9907-c7020b056821.png)

**Comment by tschams** [comment]
I have a similar issue with windows. I am getting back what you expected from curl. In vs code output, I am getting Got 0 sessions for ...

**Comment by tsgsz** [comment]
here is the result:
![image](https://user-images.githubusercontent.com/990478/227710799-5741714d-2af0-4fb4-8aca-30613aef72a5.png)

**Comment by tsgsz** [comment]
And the outputs of gitauth and main:
![image](https://user-images.githubusercontent.com/990478/227710897-7bc80094-1def-4617-a831-61d622bc1c40.png)
![image](https://user-images.githubusercontent.com/990478/227710907-e2df84f7-d632-4bea-8346-6310d87c4b8c.png)

**Comment by tsgsz** [comment]
> Ok so clearly we have a situation where the request to exchange the code for a token isn't working... So the best way to debug this is to see if the request actually goes to us...
> 
> In other words, we've back up to [#160649 (comment)](https://github.com/microsoft/vscode/issues/160649#issuecomment-1302787235) but this time with your mac instead of your windows machine.
> 
> Can you run:
> 
> ```
> curl -X POST https://vscode.dev/codeExchangeProxyEndpoints/github/login/oauth/access_token -d 'code=2345'
> ```
> 
> If it's working it should come back with `bad_verification_code: The code passed is incorrect or expired. (https://docs.github.com/apps/managing-oauth-apps/troubleshooting-oauth-app-access-token-request-errors/#bad-verification-code)`.

Guess what, Funny thing happend, I launched vscode from anaconda navigator in windows, and my login state are erased, so, My windows can't work again. LOL....

**Comment by tsgsz** [comment]
> > Ok so clearly we have a situation where the request to exchange the code for a token isn't working... So the best way to debug this is to see if the request actually goes to us...
> > In other words, we've back up to [#160649 (comment)](https://github.com/microsoft/vscode/issues/160649#issuecomment-1302787235) but this time with your mac instead of your windows machine.
> > Can you run:
> > ```
> > curl -X POST https://vscode.dev/codeExchangeProxyEndpoints/github/login/oauth/access_token -d 'code=2345'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If it's working it should come back with `bad_verification_code: The code passed is incorrect or expired. (https://docs.github.com/apps/managing-oauth-apps/troubleshooting-oauth-app-access-token-request-errors/#bad-verification-code)`.
> 
> Guess what, Funny thing happend, I launched vscode from anaconda navigator in windows, and my login state are erased, so, My windows can't work again. LOL....

there has something new：
![1679754201810](https://user-images.githubusercontent.com/990478/227722923-00bffda9-d646-4f99-a64d-d0d61c6ded6a.png)

**Comment by tsgsz** [comment]
I have a thought, is there any chance that the code for exchange token has some special character will cause this problem?

here is the url:

https://github.com/login/oauth/authorize?client_id=01ab8ac9400c4e429b23&redirect_uri=https://vscode.dev/redirect&scope=user:email&state=vscode%253A%252F%252Fvscode.github-authentication%252Fdid-authenticate%253Fnonce%253Daf2553cae834a1bd%2526windowId%253D1

**Comment by everdrone** [comment]
Having the exact same issue, so annoying

**Comment by cheestudio** [comment]
@everdrone 

I finally got to the bottom of this after using the Inspector Dev Tools in VSCode. 

It was due to a previous installation of DOCKER which set some Windows System Variables, notably this:

![image](https://user-images.githubusercontent.com/2074095/228611127-18465237-f9ab-443c-920f-80a291461f0d.png)

And an entry setting a 127.0.0.1:XXXX entry (sorry,I did not get a screenshot of that).

I removed both these entries and GitHub was able to authenticate.

So, check your system's environment variables, especially if you've used VM software like Docker! Hope this helps! 🙂

**Comment by everdrone** [comment]
@cheestudio Thanks! I'll check it out asap!

**Comment by tsgsz** [comment]
> Not sure about all of your issues exactly, what solved my issue was adding "http.proxySupport": "off" to vs code settings I believe for extensions.

I did this, succeed, thanks

**Comment by PaulinaLL** [comment]
same issue on mac

**Comment by tsgsz** [comment]
> > Not sure about all of your issues exactly, what solved my issue was adding "http.proxySupport": "off" to vs code settings I believe for extensions.
> 
> I did this, succeed, thanks

also succeed on windows. I think this is the anwser @TylerLeonhardt

**Comment by TylerLeonhardt** [comment]
@tsgsz I'm so glad! I've felt really bad that we haven't made much progress before. This is an eye opening discovery!

**Comment by TylerLeonhardt** [comment]
@tsgsz do you have a corporate proxy applied on these machines?

**Comment by tsgsz** [comment]
> 

I don't think so, the system variable is empty, and a new mac also has the problem.

**Comment by tuxslayer** [comment]
Having the same problem on macos 13.3

**Comment by TylerLeonhardt** [comment]
Can you try out @chrmarti's Proxy debugging extension here: https://github.com/microsoft/vscode-remote-release/issues/8248#issuecomment-1494127068

and let me know how it goes?

**Comment by ylht** [comment]
I guess that it maybe dns pollution, can you try to write the ip into the hosts file?

**Comment by HoniSanders** [comment]
Try going to settings/Application/Proxy:

Http: Proxy Strict SSL

- [x] Controls whether the proxy server certificate should be verified against the list of supplied CAs.

Uncheck that box!

You can see that expired certificates can lead to this problem in this thread: https://github.com/microsoft/vscode/issues/97434

**Comment by GrandStrateguerre** [comment]
Hello all,

I set it this option from override to off :
![image](https://user-images.githubusercontent.com/86523070/234520461-ba07f519-7f11-4552-b124-7f7d4c38862d.png)

And after it had worked.

Code version : 1.77.3

Thanks to all for your help :)

Best regards

**Comment by sumanth-lingappa** [comment]
> Not sure about all of your issues exactly, what solved my issue was adding "http.proxySupport": "off" to vs code settings I believe for extensions.

This solved for me as well. However @TylerLeonhardt, will we have any problem to turn-off `http.proxySupport`?

**Comment by TylerLeonhardt** [comment]
@chrmarti is the best person to answer that.

**Comment by chrmarti** [comment]
> > Not sure about all of your issues exactly, what solved my issue was adding "http.proxySupport": "off" to vs code settings I believe for extensions.
> 
> This solved for me as well. However @TylerLeonhardt, will we have any problem to turn-off `http.proxySupport`?

You won't see any issues if you are not behind a network proxy. This still suggests that there is a bug on our side.

Could you install the Network Proxy Test extension (https://marketplace.visualstudio.com/items?itemName=chrmarti.network-proxy-test) and check the output of `F1` > `Network Proxy Test: Test Connection` in VS Code?

**Comment by sumanth-lingappa** [comment]
@chrmarti You want me to send the output of the above run with or without `"http.proxySupport":"off"` ?

**Comment by chrmarti** [comment]
@sumanth-lingappa Without `"http.proxySupport":"off"` is the important one. Thanks.

**Comment by sumanth-lingappa** [comment]
> > > Not sure about all of your issues exactly, what solved my issue was adding "http.proxySupport": "off" to vs code settings I believe for extensions.
> > 
> > 
> > This solved for me as well. However @TylerLeonhardt, will we have any problem to turn-off `http.proxySupport`?
> 
> You won't see any issues if you are not behind a network proxy. This still suggests that there is a bug on our side.
> 
> Could you install the Network Proxy Test extension (https://marketplace.visualstudio.com/items?itemName=chrmarti.network-proxy-test) and check the output of `F1` > `Network Proxy Test: Test Connection` in VS Code?

```zsh
Note: Make sure to replace all sensitive information with dummy values before sharing this output.

VS Code 1.77.3 (704ed70d4fd1c6bd6342c436f1ede30d1cff4710)
Network Proxy Test 0.0.6
darwin 22.1.0 x64

Settings:
- http.proxy: 
- http.proxyAuthorization: null
- http.proxyStrictSSL: true
- http.proxySupport: override
- http.systemCertificates: true

Environment variables:

Sending GET request to https://example.com...
vscode-proxy-agent: PROXY 127.0.0.1:8009
Received error: certificate has expired (CERT_HAS_EXPIRED)
Retrying while ignoring certificate issues to collect information on the certificate chain.

Sending GET request to https://example.com (allowing unauthorized)...
vscode-proxy-agent: PROXY 127.0.0.1:8009
Received response:
- Status: 200 OK
Certificate chain:
- Subject: www.example.org (Internet Corporation for Assigned Names and Numbers)
  Subject alt: DNS:www.example.org, DNS:example.net, DNS:example.edu, DNS:example.com, DNS:example.org, DNS:www.example.com, DNS:www.example.edu, DNS:www.example.net
  Validity: Jan 13 00:00:00 2023 GMT - Feb 13 23:59:59 2024 GMT
  Fingerprint: 80:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:44
- Subject: Network Security (iboss Network Security)
  Validity: Nov 23 06:51:30 2022 GMT - Dec 25 06:51:30 2023 GMT
  Fingerprint: 23:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:FA
  Self-signed
```

**Comment by chrmarti** [comment]
@sumanth-lingappa There is a proxy configured in your OS running on your local machine at 127.0.0.1:8009. It issues its own certificates and has the CA certificate "Network Security (iboss Network Security)" registered in your OS as a root certificate. That root certificate has expired and is causing the request to fail.

You could uninstall that proxy or update its root certificate in your OS if you want to keep it running.

**Comment by chrmarti** [comment]
@sumanth-lingappa Does searching for extensions in VS Code's Extensions viewlet work? I would expect that to use the OS setting and fail too (`"http.proxySupport":"off"` won't affect that). 🤔

**Comment by chrmarti** [comment]
@sumanth-lingappa I misread the end date of the "Network Security (iboss Network Security)" certificate. It has not expired, but the first connection attempt failed with "certificate has expired". Not sure which certificate has expired. Is your system's clock set correctly (taking a guess)?

**Comment by sumanth-lingappa** [comment]
@chrmarti, yes the system clock is correct.

**Comment by chrmarti** [comment]
@sumanth-lingappa Could you use `F1` > `Network Proxy Test: Show OS Certificates` and check if that has the same "Network Security (iboss Network Security)" entry as we have seen above? Please post it here if it is different.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!
